### PR TITLE
ICNS integration

### PR DESCRIPTION
**Pull request by iboss-ptk**
Add [ICNS](https://www.icns.xyz/) integration for cosmos-kit. This PR contains
- `@cosmos-kit/icns` as a separated package
- icns hooks in `react` package, let me know if it makes more sense to separate the hook logic out as it's own package (eg. `@cosmos-kit/icns-react`) or moved to `@cosmos-kit/icns`

The goal is to make it modular and not imposed itself as a default name service so developer can choose what they want.

### CODE REVIEW

File: packages/react/package.json [hunk 1]
```diff
@@ -111,13 +111,15 @@
     "@babel/runtime": "7.11.2",
     "@chakra-ui/react": "2.4.3",
     "@cosmos-kit/core": "^0.27.13",
+    "@cosmos-kit/icns": "^0.0.1",
```
Comment (pyramation) [inline packages/react/package.json:114]: I think adding `@cosmos-kit/icns` inside of `@cosmos-kit/react` is sort of like adding it to the core... is there a way to move this logic in the `@cosmos-kit/icns` package?

Ideally we can load everything as a plugin w/o modifying `@cosmos-kit/core` or `@cosmos-kit/react`

Comment (iboss-ptk) [reply packages/react/package.json:114]: Make sense, I will move the logic!

**Comment by iboss-ptk** [comment]
Fixed according to the feedback, lmk if there is anything else to update @pyramation

File: packages/icns/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import { CosmWasmClient } from '@cosmjs/cosmwasm-stargate';
+import { useChain, useWallet } from '@cosmos-kit/react';
+import useSWR from 'swr';
```
Comment (pyramation) [inline packages/icns/src/index.ts:3]: hey, what is `swr` needed for?

File: packages/icns/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  icnsNames: IcnsNamesResponse;
+  isLoading: boolean;
+  error: unknown | undefined;
+} => {
+  const { address } = useWallet();
+  const osmosis = useChain('osmosis');
+  const { data, error, isLoading } = useSWR(
+    `${swrNamespace}/${address}`,
+    async () => {
+      const client = await osmosis.getCosmWasmClient();
```
Comment (pyramation) [inline packages/icns/src/index.ts:65]: this is the way! thanks for using the right tools for the job ;)

Comment (pyramation) [reply packages/icns/src/index.ts:65]: very soon!

File: packages/icns/src/index.ts (hunk 2 above)
Comment (iboss-ptk) [reply packages/icns/src/index.ts:3]: seems like you have already figured out :)

**Pull request merged**